app/prisma/schema.prisma

```diff
@@ -613,22 +615,70 @@ model UsageLog {
 
     inputTokens  Int
     outputTokens Int
+    inputCost    Decimal @default(0)
+    outputCost   Decimal @default(0)
     cost         Float
     type         UsageType @default(EXTERNAL)
     billable     Boolean   @default(true)
+    baseModel  String?
 
     fineTuneId String?   @db.Uuid
     fineTune   FineTune? @relation(fields: [fineTuneId], references: [id], onDelete: SetNull)
-
-    projectId String?  @db.Uuid
-    project   Project? @relation(fields: [projectId], references: [id], onDelete: Cascade)
+    projectId  String?  @db.Uuid
+    project    Project? @relation(fields: [projectId], references: [id], onDelete: Cascade)
+    invoiceId String?  @db.Uuid
+    invoice  Invoice? @relation(fields: [invoiceId], references: [id], onDelete: SetNull)
 
     createdAt DateTime @default(now())
 
     @@index([fineTuneId, createdAt, type])
     @@index([projectId, createdAt, type])
 }
 
+enum InvoiceStatus {
+    PENDING
+    PAID
+    CANCELLED
+    REFUNDED
+}
+model Invoice {
+    id String @id @default(uuid()) @db.Uuid
+    amount         Decimal @default(0)
+    status         InvoiceStatus @default(PENDING)
+    slug           String
+    billingPeriod  String? // MM YYYY
+    paidAt         DateTime?
+
+    description Json? // Array of objects: { [key: string]: string }
+    paymentId String?
+
+    remainingCredits Decimal?
```

Comment:
Did we decide that we no longer need this field?

Reply:
Apologies for the belated review. Yes, I believe we can do without this field based on the way the `CreditAdjustment` table works now.